src/tyr/isochrone_serializer.cc

```diff
@@ -142,5 +142,66 @@ std::string serializeIsochrones(const Api& request,
 
   return ss.str();
 }
+std::string serializeIsochronePbf(Api& request,
+                                  std::vector<midgard::GriddedData<2>::contour_interval_t>& intervals,
+                                  midgard::GriddedData<2>::contours_t& contours) {
+  // construct pbf output
+  Isochrone& isochrone = *request.mutable_isochrone();
+
+  // construct contours
+  for (size_t isoline_index = 0; isoline_index < contours.size(); ++isoline_index) {
+    const auto& contour = contours[isoline_index];
+    const auto& interval = intervals[isoline_index];
+
+    valhalla::Isochrone::Contour con;
+    con.set_metric(std::get<2>(interval) == "time" ? Isochrone_metric_type_time
+                                                   : Isochrone_metric_type_distance);
+
+    con.set_metric_value(std::get<1>(interval));
+
+    // for each feature
+    for (const auto& feature : contour) {
+      // for each ring/contour
+      for (const auto& ring : feature) {
+        // construct a geometry
+        Isochrone_Geometry geom;
+        for (const auto& pair : ring) {
+          geom.add_coords(round(pair.lng() * 1e6));
+          geom.add_coords(round(pair.lat() * 1e6));
+        }
+        con.add_geometries()->CopyFrom(geom);
```

Comment:
not sure if `CopyFrom` is best here.. sounds like it's copying;) there's also a `SwapSmth` function(s) or change the order of things by creating a contour at the top with `auto* con = isochrone.mutable_contours->Add()`  and then a `auto* geom = con.mutable_geometries().Add()` where you can do `geom->add_coords()`.